src/realm/table.cpp

```diff
@@ -1710,6 +1710,10 @@ TableRef Table::get_link_target(ColKey col_key) noexcept
 
 size_t Table::count_int(ColKey col_key, int64_t value) const
 {
+    if (auto index = this->get_search_index(col_key)) {
```

Comment:
@jedelbo How are these changes in table.cpp related to this PR?

Reply:
If those fixes were not done, one of the tests would take forever.

Reply:
So these are optimizations that affects the end user as well?
Should they not be done for the other types?
Are they only relevant to the count aggregation or others as well?

Reply:
All optimizations somehow affects the user.
They are added for types that support index.
Only relevant for count.

Reply:
@jedelbo Shouldn't there be added a small note about the enhancement then? I assume it's not insignificant if it was taking the test to a crawl?

Reply:
If this was something we had released - yes.

Reply:
Good point :-)
Do we have or should we have benchmarks that detect this?

Reply:
This is a somewhat exotic function. I would expect applications rather to use a query for this functionality.